pkg/prometheus/operator.go

```diff
@@ -472,21 +497,14 @@ func (c *Operator) syncNodeEndpoints() error {
 		return errors.Wrap(err, "listing nodes failed")
 	}
 
-	for _, n := range nodes.Items {
-		address, _, err := nodeAddress(n)
-		if err != nil {
-			return errors.Wrapf(err, "failed to determine hostname for node (%s)", n.Name)
+	addresses, errs := getNodeAddresses(nodes)
+	if len(errs) > 0 {
+		for _, err := range errs {
+			level.Warn(c.logger).Log("err", err)
+			c.nodeAddressLookupErrors.Inc()
```

Comment:
this should be `c.nodeAddressLookupErrors.Add(float64(len(errs)))`

Reply:
Interesting, I started with that, then got a comment from a colleague that it was hard to read. Figured that for something run every 3 mins or so it wouldn't matter performance-wise.

Changed it back, but would be interesting to know your perspective on this over the other? Performance, readability, something else?